Firestore/Swift/Source/Codable/DocumentID.swift

```diff
@@ -101,9 +105,12 @@ public struct DocumentID<Value: DocumentIDWrappable & Codable>:
   // MARK: - `Codable` implementation.
 
   public init(from decoder: Decoder) throws {
-    throw FirestoreDecodingError.decodingIsNotSupported(
-      "DocumentID values can only be decoded with Firestore.Decoder"
-    )
+    guard let reference = decoder.userInfo[documentRefUserInfoKey] as? DocumentReference else {
+      throw FirestoreDecodingError.decodingIsNotSupported(
+        "Could not find DocumentReference for user info key: \(documentRefUserInfoKey)"
+      )
+    }
+    try self.init(from: reference)
```

Comment:
What's the motive behind this behavioral change? I'm ok with it, but it'd be good to keep an explanation in the change history.

Reply:
@morganchen12 @andrewheard @mortenbekditlevsen @ncooke3 I'm going to resolve in order to merge today, but would appreciate a resolution/ follow up PR in the next few days

Reply:
@morganchen12 The two behaviour changes I'm aware of are:
1. Developers could decode with a custom `Decoder` as long as it populates `userInfo[documentRefUserInfoKey]`, whereas before it would always throw
2. The error message has changed.

I'm wondering if we should add `; DocumentID values should only be decoded with Firestore.Decoder` for now.

That said, I'm not sure about the motive for this behavioural change.